webui/src/lib/hooks/storageConfig.tsx

```diff
@@ -0,0 +1,69 @@
+import React, {
+  FC,
+  useContext,
+  useState,
+  createContext,
+  useEffect,
+} from "react";
+
+import { config } from "../api";
+
+type StorageConfigContextType = {
+  error: Error | null;
+  loading: boolean;
+  blockstore_namespace_ValidityRegex: string | null;
+  blockstore_namespace_example: string | null;
+  blockstore_type: string | null;
+  default_namespace_prefix: string | null;
+  import_support: boolean;
+  pre_sign_support: boolean;
+  pre_sign_support_UI: boolean;
+};
+
+const storageConfigInitialState: StorageConfigContextType = {
+  error: null,
+  loading: true,
+  blockstore_namespace_ValidityRegex: null,
+  blockstore_namespace_example: null,
+  blockstore_type: null,
+  default_namespace_prefix: null,
+  import_support: false,
+  pre_sign_support: false,
+  pre_sign_support_UI: false,
+};
+
+export const fetchStorageConfig = async () => {
+  const storageConfig = await config.getStorageConfig();
+  return storageConfig;
+};
+
+const StorageConfigContext = createContext<StorageConfigContextType>(
+  storageConfigInitialState
+);
+
+export const useStorageConfig = () => {
+  const config = useContext(StorageConfigContext);
+  return config;
+};
+
+export const StorageConfigProvider: FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [storageConfig, setStorageConfig] = useState<StorageConfigContextType>(
+    storageConfigInitialState
+  );
+
+  useEffect(() => {
+    const fetchStorageConfigAndSetState = async () => {
+      const storageConfig = await fetchStorageConfig();
+      setStorageConfig(storageConfig);
```

Comment:
This is a bit nasty, because we _implicitly claim_ to pass a StorageConfigContextType but never actually verify the type.  Not sure I have a quick fix.

Reply:
It's true; it is, and we don't. Usually, when I do this, I comment that we'll refactor this as soon as we add types to the API client. I'll do this here as well.